src/services/XmppSocket.js

```diff
@@ -283,6 +283,20 @@ export default {
       this.context.$store.setContactPresence({ jid: fullJid.bare, presence: presence.show, status: presence.status })
     })
 
+    // listen for retracted messages
+    this.client.on('messageRetracted', (retracted) => {
+      this.context.$store.updateMessage({
+        stanzaId: retracted.stanzaId,
+        // replace body and links
+        body: `Moderated by ${retracted.by.resource} (${retracted.reason})`,
+        links: [],
```

Comment:
I clean `body` and `links` attributes, but may be we have to clean others?

Reply:
As I said above, I can only guess what some of these attributes correspond to but not all. I guess it is fine?

Reply:
In the app, a message contains the following attribute:

- id: attribute id of message,
- stanzaId: attribute id of `stanza-id` child node,
- from,
- to,
- **body**,
- delay: original timestamp,
- **links**: URL of XEP-0066,
- status: set when sent error or retracted occurs.

I do not feel required to clean the `from` Jid for example.

Reply:
Hmm, maybe I'm going too far but it's not impossible the nickname of the user is also used to convey harmful ideas. Maybe the `from` could be replaced as well? Even though it's probably not a requirement.

Reply:
It cost very few, let's go.
Do I replace nick with some asterisks?

Reply:
I speak to fast... if I change the entire `from` object it break the display (as the message is not attached to the room), and update only the local (nick) require to fetch the message before.
I propose to leave it there and if you need it later, we will handle it.
Sorry.